examples/with_winit/src/stats.rs

```diff
@@ -0,0 +1,183 @@
+        samples: T,
+        vsync: bool,
+    ) where
+        T: Iterator<Item = &'a u64>,
+    {
+        let width = (viewport_width * 0.4).max(200.).min(600.);
+        let height = width * 0.7;
+        let x_offset = viewport_width - width;
+        let y_offset = viewport_height - height;
+        let offset = Affine::translate((x_offset, y_offset));
+
+        // Draw the background
+        sb.fill(
+            Fill::NonZero,
+            offset,
+            &Brush::Solid(Color::rgba8(0, 0, 0, 200)),
+            None,
+            &Rect::new(0., 0., width, height),
+        );
+
+        let labels = [
+            format!("Frame Time: {:.2} ms", self.frame_time_ms),
+            format!("Frame Time (min): {:.2} ms", self.frame_time_min_ms),
+            format!("Frame Time (max): {:.2} ms", self.frame_time_max_ms),
+            format!("VSync: {}", if vsync { "on" } else { "off" }),
+            format!("Resolution: {viewport_width}x{viewport_height}"),
+        ];
+
+        // height / 2 is dedicated to the text labels and the rest is filled by the bar graph.
+        let text_height = height * 0.5 / (1 + labels.len()) as f64;
+        let left_margin = width * 0.01;
+        let text_size = (text_height * 0.9) as f32;
+        for (i, label) in labels.iter().enumerate() {
+            text.add(
+                sb,
+                None,
+                text_size,
+                Some(&Brush::Solid(Color::WHITE)),
+                offset * Affine::translate((left_margin, (i + 1) as f64 * text_height)),
+                &label,
+            );
+        }
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((width * 0.67, text_height)),
+            &format!("FPS: {:.2}", self.fps),
+        );
+
+        // Plot the samples with a bar graph
+        use PathEl::*;
+        let graph_max_height = height * 0.5;
+        let graph_max_width = width - 2. * left_margin;
+        let bar_extent = graph_max_width / (SLIDING_WINDOW_SIZE as f64);
+        let bar_width = bar_extent * 0.3;
+        let bar = [
+            MoveTo((0., graph_max_height).into()),
+            LineTo((0., 0.).into()),
+            LineTo((bar_width, 0.).into()),
+            LineTo((bar_width, graph_max_height).into()),
+        ];
+        for (i, sample) in samples.enumerate() {
+            let t = offset * Affine::translate((i as f64 * bar_extent, graph_max_height));
+            // The height of each sample is based on its ratio to the maximum observed frame time.
+            // Currently this maximum scale is sticky and a high temporary spike will permanently
+            // shrink the draw size of the overall average sample, so scale the size non-linearly to
+            // emphasize smaller samples.
+            let h = (*sample as f64) * 0.001 / self.frame_time_max_ms;
+            let s = Affine::scale_non_uniform(1., -h.sqrt());
```

Comment:
This logic is non-obvious to me; I suspect a maximum height based on the maximum seen last render might be better. I'd much rather keep things linear.

Reply:
I switched this to linear now. I thought using sqrt here added value initially but now that we can reset the max/min frame times with a keypress (which resets the scale) this type of visualization is not necessary.

I'll follow up with the dashed lines for 16.6ms/33.3ms as you suggested, which should provide a good relative visual anchor.